pkg/controller/replicaset/replica_set.go

```diff
@@ -219,13 +219,7 @@ func (rsc *ReplicaSetController) getReplicaSetsWithSameController(rs *apps.Repli
 		}
 	}
 
-	if klog.V(2).Enabled() {
-		var relatedNames []string
-		for _, r := range relatedRSs {
-			relatedNames = append(relatedNames, r.Name)
-		}
-		klog.InfoS("Found related ReplicaSets", "replicaSet", klog.KObj(rs), "relatedReplicaSets", relatedNames)
-	}
+	klog.V(2).InfoS("Found related ReplicaSets", "replicaSet", klog.KObj(rs), "relatedReplicaSets", klog.KObjs(relatedRSs))
```

Comment:
`klog.KObjs` does not produce exactly the same output but that should be fine because it works better with JSON.

I am worried that it might cause overhead when the log message is not actually emitted (see https://github.com/kubernetes/kubernetes/issues/106945).

Therefore here the prior "enabled" check might make sense even when using `klog.KObjs`.

Reply:
Seems definitely a valid concern, should we try to prioritize the optimization before we push this out? It definitely could have a large impact in terms of resource consumption

Reply:
In other places `klog.KObjs` was used without outer if check. But I myself also lean towards keeping the check until `klog.KObj` can shown to not have a performance overhead when called unconditionally.